vision/mlp_mnist/mlp_mnist.jl

```diff
@@ -131,21 +134,21 @@ function train(; kws...)
     ## Create test and train dataloaders
     train_loader, test_loader = getdata(args)
 
+    @info "Constructing model and starting training"
     ## Construct model
     model = build_model() |> device
-    ps = Flux.params(model) ## model's trainable parameters
-    
+
     ## Optimizer
-    opt = ADAM(args.η)
-    
+    opt = Flux.setup(ADAM(args.η), model)
+
     ## Training
     for epoch in 1:args.epochs
         for (x, y) in train_loader
             x, y = device(x), device(y) ## transfer data to device
-            gs = gradient(() -> logitcrossentropy(model(x), y), ps) ## compute gradient
-            Flux.Optimise.update!(opt, ps, gs) ## update parameters
+            gs = gradient(model -> loss(model, x, y), model) ## compute gradient
```

Comment:
I think it's helpful to use different variable names for the local variable in the function:

```suggestion
            gs = gradient(m -> loss(m, x, y), model) ## compute gradient
```
Alternatively, having defined this function `loss`, you can just call it:
```suggestion
            gs = gradient(loss, model, x, y)  ## compute gradient
```

Reply:
Or if you delete `loss` definition:
```suggestion
            gs = gradient(m -> logitcrossentropy(m(x), y), model)  ## compute gradient of the loss
```

Reply:
Thanks, @mcabbott.

Cf. https://github.com/FluxML/model-zoo/pull/385#discussion_r1059776647